docs/authorization-introduction.md

```diff
@@ -0,0 +1,124 @@
+The access control management evolved from the need to mitigate the risk of unauthorized access to data.
+
+Several control models were built during time to approach the various access control needs: Access Control Lists, Role Based Access Control, Attribute Based Access Control, Policy Based Access Control, Risk Adaptive Access Control. A summary of these models is made on [Survey on access control models][], the document offers a description of the PBAC model that Udaru uses.
+
+Udaru is based on the [node-pbac][] module, module that as the developer claims is inspired and mostly compatible with the [AWS IAM][] policy engine.
+
+Altough the [AWS IAM Policy][] is not the documentation for [node-pbac][] it can be used as reference for details regarding policies and policy elements.
+
+## The Authorization model
+
+An access control model can be built using the **Organization**, **Team** and **User** entities. The Organizations can have Teams or Users attached to them. Teams can have sub Teams or Users attached to them. Only the Team elements can have Team elements (sub Teams) attached to it.
+
+Policies can be attached to Organizations, Teams and Users. The policy engine evaluates all policies from all levels when establishing if a User can perform an Action over a Resource.
+
+**Important note:** Teams, Users and Policies can't be shared across Organizations.
+
+A policy is a document that describes what type of access the policy owner has to a resource or a group of resources. Given the policies, the action to perform and a resource, the engine returns `true` if the action is authorized, `false` otherwise.
+
+A User can have policies attached to himself or inherit them from his Team(s) and Organization.
+
+A policy looks like:
+
+```javascript
+{
+  id: 'Policy ID',
+  version: '2016-07-01',
+  name: 'Policy name',
+  statements: { Statement: [{
+    Effect: 'Allow',
+    Action: ['Documents:Read'],
+    Resource: ['wonka:documents:/public/*']
+  }] }
+}
+```
+
+The main elements of a policy are:
+- The policy ID,
+- The policy version,
+- The policy name,
+- The policy statements that contains an array of statements.
+
+The elements of a Statement are:
+- The Action to be performed. Example: `'authorization:teams:create'` or `'authorization:organizations:*'`,
+- The Resource on which the action is performed. Example: `'FOO:orga:CLOUDCUCKOO:scenario:*:entity:north-america-id'`,
+- The Effect - has the value 'Allow' or the value 'Deny'.
+
+Wildcards can be used in Action and Resource names.
+
+## The authorization system structure
+
+The Udaru authorization system uses the same PBAC approach to perform administration of its internal resources as it does for controlling access to external resources. More details can be seen on [Authorization Technical Specs][]
+
+The exposed functionality endpoints are structured as the following:
+- Routes used to validate or not if actions over resources are authorized, these are exposed as [Public routes][],
```

Comment:
`Routes used to validate if actions over resources are authorized are exposed as [Public routes]`